File: vk-checker/src/main/java/com/merfemor/vkwallwatcher/vk/NamedRunnable.java
package com.merfemor.vkwallwatcher.vk;

public abstract class NamedRunnable implements Runnable {
    private final String threadNamePrefix;

    public NamedRunnable(String threadNamePrefix) {
        this.threadNamePrefix = threadNamePrefix;
    }

    @Override
    public final void run() {
        Thread thread = Thread.currentThread();
        String oldName = thread.getName();
        thread.setName(threadNamePrefix + "_" + oldName);
        execute();
        thread.setName(oldName);
    }

    protected abstract void execute();
}

File: vk-checker/src/main/java/com/merfemor/vkwallwatcher/vk/NamedRunnable.kt
package com.merfemor.vkwallwatcher.vk

fun Runnable.named(threadNamePrefix: String): Runnable {
    val runnable = this
    return object : NamedRunnable(threadNamePrefix) {
        override fun execute() {
            runnable.run()
        }
    }
}
